File: common_py/input_access.py
import os
import requests

from pathlib import Path  # Python 3.6+ only
from dotenv import load_dotenv

env_path = Path('.') / '.env'
load_dotenv(dotenv_path=env_path)

access_token = os.getenv("ACCESS_TOKEN")

if access_token is None:
  raise Exception("Access token not found")

def fetch_input(day) -> str:
  cache_dir = "./inputCache"

  if not os.path.exists(cache_dir):
    os.makedirs(cache_dir)
  
  cache_file_path = f"{cache_dir}/day{day}.txt"

  if os.path.exists(cache_file_path):
    file = open(cache_file_path)
    content = file.read()
    file.close()
    return content


  uri = f"https://adventofcode.com/2020/day/{day}/input"
  cookies = {'session': access_token}
  response = requests.get(uri, cookies=cookies)

  input = response.text

  file = open(cache_file_path, mode="w")
  file.write(input)
  file.close()

  return input

if __name__ == "__main__":
    print(fetch_input(1))

File: common_ts/inputAccess.ts
import { config } from "https://deno.land/x/dotenv/mod.ts";

const accessToken = config().ACCESS_TOKEN;

export const fetchInput = async (day: number): Promise<string> => {
  const cacheDir = "./inputCache";

  if (!(await exists(cacheDir))) {
    await Deno.mkdir(cacheDir);
  }

  const cachePath = `${cacheDir}/day${day}.txt`;
  if (await exists(cachePath)) {
    return (await Deno.readTextFile(cachePath));
  }

  console.log("Input not found in cache, downloading from aoc.com...");

  const uri = `https://adventofcode.com/2020/day/${day}/input`;

  const headers = new Headers();
  headers.set("Cookie", `session=${accessToken}`);

  const response = await fetch(uri, { credentials: "include", headers });

  const input = await response.text();

  await Deno.writeTextFile(cachePath, input, { create: true });
  return input;
};

const exists = async (path: string): Promise<boolean> => {
  try {
    await Deno.stat(path);
    return true;
  } catch (error) {
    if (error instanceof Deno.errors.NotFound) {
      return false;
    }
    throw error;
  }
};
